client/src/client.rs

```diff
@@ -193,15 +211,23 @@ struct ResponseReport(eyre::Report);
 
 impl ResponseReport {
     /// Constructs report with provided message
-    fn with_msg<S>(msg: S, response: &Response<Vec<u8>>) -> Self
+    ///
+    /// # Errors
+    /// If response body isn't valid utf-8 string
+    fn with_msg<S>(msg: S, response: &Response<Vec<u8>>) -> Result<Self, Self>
     where
         S: AsRef<str>,
     {
         let status = response.status();
-        let body = String::from_utf8_lossy(response.body());
+        let body = std::str::from_utf8(response.body());
         let msg = msg.as_ref();
 
-        Self(eyre!("{msg}; status: {status}; response body: {body}"))
+        body.map_err(|_| {
+            Self(eyre!(
+                "{msg}; status: {status}; body isn't valid utf-8 string"
```

Comment:
```suggestion
                "{msg}; status: {status}; body isn't a valid utf-8 string"
```

@outoftardis , am I right?

Reply:
You're correct.